Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceSolver.H

```diff
@@ -206,127 +207,128 @@ class FiniteDifferenceSolver
+            amrex::MultiFab* Gfield,
+            ablastr::fields::VectorField const& Bfield,
             amrex::Real dt);
 
         void EvolveRhoCartesianECT (
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& Efield,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& edge_lengths,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& face_areas,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 >& ECTRhofield, int lev);
+            ablastr::fields::VectorField const& Efield,
+            ablastr::fields::VectorField const& edge_lengths,
+            ablastr::fields::VectorField const& face_areas,
+            ablastr::fields::VectorField const& ECTRhofield, int lev);
 
         void EvolveBCartesianECT (
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Bfield,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& face_areas,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& area_mod,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 >& ECTRhofield,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Venl,
+            ablastr::fields::VectorField const& Bfield,
+            ablastr::fields::VectorField const& face_areas,
+            ablastr::fields::VectorField const& area_mod,
+            ablastr::fields::VectorField const& ECTRhofield,
+            ablastr::fields::VectorField const& Venl,
             std::array< std::unique_ptr<amrex::iMultiFab>, 3 >& flag_info_cell,
             std::array< std::unique_ptr<amrex::LayoutData<FaceInfoBox> >, 3 >& borrowing,
             int lev, amrex::Real dt
         );
 
         template< typename T_Algo >
         void ComputeDivECartesian (
-            const std::array<std::unique_ptr<amrex::MultiFab>,3>& Efield,
+            ablastr::fields::VectorField const & Efield,
             amrex::MultiFab& divE );
 
         template< typename T_Algo, typename T_MacroAlgo >
         void MacroscopicEvolveECartesian (
-            std::array< std::unique_ptr< amrex::MultiFab>, 3>& Efield,
-            std::array< std::unique_ptr< amrex::MultiFab>, 3> const& Bfield,
-            std::array< std::unique_ptr< amrex::MultiFab>, 3> const& Jfield,
-            std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& edge_lengths,
+            ablastr::fields::VectorField const& Efield,
+            ablastr::fields::VectorField const& Bfield,
+            ablastr::fields::VectorField const& Jfield,
+            ablastr::fields::VectorField const& edge_lengths,
             amrex::Real dt,
             std::unique_ptr<MacroscopicProperties> const& macroscopic_properties);
 
         template< typename T_Algo >
         void EvolveBPMLCartesian (
             std::array< amrex::MultiFab*, 3 > Bfield,
-            std::array< amrex::MultiFab*, 3 > Efield,
+            ablastr::fields::VectorField const Efield,
```

Comment:
Clang-tidy error "parameter 'Efield' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions [readability-avoid-const-params-in-decls,-warnings-as-errors]":
```suggestion
            ablastr::fields::VectorField Efield,
```